core/modules/nat.cc

```diff
@@ -54,14 +54,28 @@ using bess::utils::UpdateChecksumWithIncrement;
 using bess::utils::UpdateChecksum16;
 
 CommandResponse NAT::Init(const bess::pb::NATArg &arg) {
-  for (const std::string &ext_addr : arg.ext_addrs()) {
+  for (const auto &address_range : arg.ext_addrs()) {
+    auto ext_addr = address_range.ext_addr();
     be32_t addr;
+
     bool ret = bess::utils::ParseIpv4Address(ext_addr, &addr);
     if (!ret) {
       return CommandFailure(EINVAL, "invalid IP address %s", ext_addr.c_str());
     }
 
     ext_addrs_.push_back(addr);
+    // Add a port range list
+    std::list<PortRange> port_list;
+    for (const auto &range : address_range.port_ranges()) {
```

Comment:
it would be nice to do simple sanity check here, (e.g., `start <= end`, 16-bit port numbers, etc.)